crates/interpreter/src/valid.rs

```diff
@@ -742,15 +779,23 @@ impl<'a, 'm> Expr<'a, 'm> {
         self.for_each(expected, |x, y| check(x.matches(y)))
     }
 
-    fn push_label(&mut self, type_: FuncType<'m>, kind: LabelKind) -> CheckResult {
+    fn push_label(
+        &mut self, type_: FuncType<'m>, kind: LabelKind,
+        side_table_log_entries: Vec<(&'m [u8], usize)>,
+    ) -> CheckResult {
         self.pops(type_.params)?;
         let stack = type_.params.iter().cloned().map(OpdType::from).collect();
-        let label = Label { type_, kind, polymorphic: false, stack };
+        let label = Label { type_, kind, polymorphic: false, stack, side_table_log_entries };
         self.labels.push(label);
         Ok(())
     }
 
     fn end_label(&mut self) -> CheckResult {
+        for (label_parser_pos, idx) in &self.labels.last().unwrap().side_table_log_entries {
```

Comment:
Let's just take the branches since we're gonna trash the label anyway.

```rust
let branches = core::mem::take(&mut self.label().branches);
for branch in branches {
```

Maybe we can inline, but I'm not sure the borrow checker will see it's fine.